joseTOV/eos.py

```diff
@@ -111,135 +117,350 @@ class MetaModel_EOS_model(Interpolate_EOS_model):
         
-        # Initialize with parent class
-        super().__init__(ns, ps, es)
+        # t_sat or TFGsat is the kinetic energy per nucleons in SM and at saturation, see just after eq (13) in the margueron paper
+        self.t_sat = 3 * utils.hbar ** 2 / (10 * utils.m) * (3 * jnp.pi ** 2 * self.nsat / 2) ** (2/3)
         
-        # Use ns rather than self.n because of unit conversion in super init
-        self.cs2 = self.cs2_from_number_density_nuclear_unit(ns)
-
-    def esym(self, n: Float[Array, "n_points"]):
-        x = (n - self.nsat) / (3.0 * self.nsat)
+        # Potential energy 
+        # v_sat is defined in equations (22) - (26) in the Margueron et al. paper
+        # TODO: there are more terms here, perhaps check the other reference that Rahul shared?
+        v_sat_0 = self.E_sat -     self.t_sat * ( 1 +   self.kappa_sat +   self.kappa_sat2 +     self.kappa_sat3)
+        v_sat_1 =            -     self.t_sat * ( 2 + 5*self.kappa_sat + 8 * self.kappa_sat2 + 11* self.kappa_sat3)
+        v_sat_2 = self.K_sat - 2 * self.t_sat * (-1 + 5*self.kappa_sat + 20*self.kappa_sat2 + 44* self.kappa_sat3)
+        v_sat_3 = self.Q_sat - 2 * self.t_sat * ( 4 - 5*self.kappa_sat + 40*self.kappa_sat2 + 220* self.kappa_sat3)
+        v_sat_4 = self.Z_sat - 8 * self.t_sat * (-7 + 5*self.kappa_sat - 10*self.kappa_sat2 + 110* self.kappa_sat3) 
+        
+        self.v_sat = jnp.array([v_sat_0, v_sat_1, v_sat_2, v_sat_3, v_sat_4])
+        
+        # v_sym2 is defined in equations (27) to (31) in the Margueron et al. paper
+        v_sym2_0 = self.E_sym -     self.t_sat * ( 2**(2/3)*( 1+  self.kappa_NM+   self.kappa_NM2+    self.kappa_NM3) - ( 1+  self.kappa_sat+   self.kappa_sat2+   self.kappa_sat3)  ) - v_nq[0]
+        v_sym2_1 = self.L_sym -     self.t_sat * ( 2**(2/3)*( 2+5*self.kappa_NM+8* self.kappa_NM2+ 11*self.kappa_NM3) - ( 2+5*self.kappa_sat+8* self.kappa_sat2+11*self.kappa_sat3)  ) - v_nq[1]
+        v_sym2_2 = self.K_sym - 2 * self.t_sat * ( 2**(2/3)*(-1+5*self.kappa_NM+20*self.kappa_NM2+ 44*self.kappa_NM3) - (-1+5*self.kappa_sat+20*self.kappa_sat2+44*self.kappa_sat3)  ) - v_nq[2]
+        v_sym2_3 = self.Q_sym - 2 * self.t_sat * ( 2**(2/3)*( 4-5*self.kappa_NM+40*self.kappa_NM2+ 220*self.kappa_NM3) - ( 4-5*self.kappa_sat+40*self.kappa_sat2+220*self.kappa_sat3) ) - v_nq[3]
+        v_sym2_4 = self.Z_sym - 8 * self.t_sat * ( 2**(2/3)*(-7+5*self.kappa_NM-10*self.kappa_NM2+ 110*self.kappa_NM3) - (-7+5*self.kappa_sat-10*self.kappa_sat2+110*self.kappa_sat3) ) - v_nq[4]
+        
+        self.v_sym2 = jnp.array([v_sym2_0, v_sym2_1, v_sym2_2, v_sym2_3, v_sym2_4])
+        
+        # Load and preprocess the crust
+        ns_crust, ps_crust, es_crust = load_crust(crust)
+        max_n_crust = max_n_crust_nsat * nsat
+        mask = ns_crust <= max_n_crust
+        ns_crust, ps_crust, es_crust = ns_crust[mask], ps_crust[mask], es_crust[mask]
+        
+        # FIXME: remove this once we discussed about this with Rahul
```

Comment:
The value is hard-coded here to be the mu_lowest as returned from the crust file that Rahul uses. Changing this value seems to significantly affect the results.